src/main/java/io/vertx/core/Future.java

```diff
@@ -617,4 +618,35 @@ static <T> Future<T> fromCompletionStage(CompletionStage<T> completionStage, Con
     });
     return promise.future();
   }
+
+  /**
+   * Park the current thread until the {@code future} is completed, when the future
+   * is completed the thread is un-parked and
+   *
+   * <ul>
+   *   <li>the result value is returned when the future was completed with a result</li>
+   *   <li>otherwise, the failure is thrown</li>
+   * </ul>
+   *
+   * @param future the future to await
+   * @return the result
+   */
+  static <T> T await(Future<T> future) {
```

Comment:
Just to be sure: what's the behaviour if I call `await` on Java 17? Shall it even be any good idea? Shall it be documented?

Reply:
right there should be more explanation about this